### PR TITLE
learn: add hasura backend course

### PR DESCRIPTION
**Pull request by praveenweb**
### Description

This PR adds the Hasura backend course based on the schema of realtime todo app. 
The course covers
- Deploying Hasura
- Data modelling
- Relationships / Foreign keys
- Views
- Authorization 
- Authentication with Auth0
- Custom Business Logic with remote schemas and event trigger

### Affected components 

- Community Content

### CODE REVIEW

**Comment by shahidhk** [comment]
Can we move all images to a CDN?

Regards,

Shahidh

**Comment by praveenweb** [comment]
@shahidhk - images have been moved to CDN.

**Comment by mnlbox** [comment]
@praveenweb , @shahidhk Guys I have some terrible with CDN data on google storage :disappointed: 
Like this issue: #521
Please provide an alternative CDN for others like me as a plan B. (If google CDN not working load from other CDN or storage)

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: "Custom Business Logic"
+---
+
+Hasura gives you CRUD + realtime GraphQL APIs with authorization & access control. However there are cases where you would want to query something which is not in your database.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md]: Second line should say: However, there are case where you would want to add custom/business logic in your app. Some examples are ....

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+Custom business logic can be handled in two ways using Hasura:
+- Writing custom GraphQL resolvers and adding it as a remote schema.
+- Creating event triggers that capture events happening on the specified tables.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md]: The intent is buried under too much jargon. It should ideally say:

'After a mutation, trigger a webhook'.

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+In the todo app backend that you have built, there are use-cases to query data which is not in the database. For example,
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic.md]: In the todo app backend that you have built, there are certain custom functionalities you may want to add:

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/1-remote-schemas.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+title: "Write custom resolvers"
+---
+
+Let's take the first use-case of fetching profile information from Auth0.
+
+Ideally you would want to maintain a single GraphQL endpoint for all your data requirements.
+
+Hasura has the ability to merge remote GraphQL schemas and provide a unified GraphQL API. To handle the use-case of fetching Auth0 profile information, we will write custom resolvers in a custom GraphQL server. Hasura can then merge this remote schema with the existing auto-generated schema.
+
+![Remote schema architecture](https://docs.hasura.io/1.0/_images/remote-schemas-arch1.png)
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/1-remote-schemas.md]: this image is huge. It doesn't fit in one fold.

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/1-remote-schemas.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
+---
+title: "Write custom resolvers"
+---
+
+Let's take the first use-case of fetching profile information from Auth0.
+
+Ideally you would want to maintain a single GraphQL endpoint for all your data requirements.
+
+Hasura has the ability to merge remote GraphQL schemas and provide a unified GraphQL API. To handle the use-case of fetching Auth0 profile information, we will write custom resolvers in a custom GraphQL server. Hasura can then merge this remote schema with the existing auto-generated schema.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/1-remote-schemas.md]: The term "remote schema" is introduced abruptly.

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
+---
+title: "Write event webhook"
+---
+
+Now let's move to the second use-case of Sending an email when a user registers on the app.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md]: Sending should be lower-case

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+When the user registers on the app using Auth0, we insert a new row into the `users` table to keep the user data in sync. Remember the rule we wrote to make a mutation?
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md]: Remember the rule we wrote during sign-up?

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+This is an `insert` operation on table `users`.
+
+Now we are going to capture this insert operation to trigger our event.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/3-write-event-webhook.md]: Add a second line which says: the payload for each event is mentioned here: <doc-link>

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/4-create-event-trigger.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
+---
+title: "Create Event Triggers"
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/4-create-event-trigger.md]: Create Event Trigger?

File: community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/4-create-event-trigger.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+To test this, we need to insert a new row into users table.
+
+Head to Console -> Data -> users -> Insert Row and insert a new row.
+
+Now head to Events tab and click on `send_email` event to browse the processed events.
+
+![Test event trigger](https://storage.googleapis.com/graphql-engine-cdn.hasura.io/learn-hasura/assets/graphql-hasura/test-event-trigger.png)
+
+Now everytime a new row is inserted into `users` table this event would be triggered.
```
Comment (tirumaraiselvan) [inline community/learn/hasura/graphql-tutorial-backend/tutorial-site/content/custom-business-logic/4-create-event-trigger.md]: this event would be triggered. -> this event trigger would be invoked.

**Review by vashisth00 — COMMENTED** [review]
Why does the proxy edit happen here ?

**Pull request merged**